internal/database/database.go

```diff
@@ -202,16 +202,89 @@ func (d *CosmosDBClient) DeleteClusterDoc(ctx context.Context, resourceID *azcor
 }
 
 func (d *CosmosDBClient) GetNodePoolDoc(ctx context.Context, resourceID *azcorearm.ResourceID) (*NodePoolDocument, error) {
-	panic("implement me")
+	// Make sure lookup keys are lowercase.
+	key := strings.ToLower(resourceID.String())
+	pk := azcosmos.NewPartitionKeyString(resourceID.SubscriptionID)
+
+	container, err := d.client.NewContainer(nodePoolsContainer)
+	if err != nil {
+		return nil, err
+	}
+
+	query := "SELECT * FROM c WHERE c.key = @key"
+	opt := azcosmos.QueryOptions{
+		PageSizeHint:    1,
+		QueryParameters: []azcosmos.QueryParameter{{Name: "@key", Value: key}},
+	}
+
+	queryPager := container.NewQueryItemsPager(query, pk, &opt)
+
+	var doc *NodePoolDocument
+	for queryPager.More() {
+		queryResponse, err := queryPager.NextPage(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, item := range queryResponse.Items {
+			err = json.Unmarshal(item, &doc)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+	if doc != nil {
+		return doc, nil
+	}
+	return nil, ErrNotFound
 }
 
 func (d *CosmosDBClient) SetNodePoolDoc(ctx context.Context, doc *NodePoolDocument) error {
-	panic("implement me")
+	// Make sure lookup keys are lowercase.
+	doc.Key = strings.ToLower(doc.Key)
+	doc.PartitionKey = strings.ToLower(doc.PartitionKey)
+
+	data, err := json.Marshal(doc)
+	if err != nil {
+		return err
+	}
+
+	container, err := d.client.NewContainer(nodePoolsContainer)
+	if err != nil {
+		return err
+	}
+
+	_, err = container.UpsertItem(ctx, azcosmos.NewPartitionKeyString(doc.PartitionKey), data, nil)
```

Comment:
todo: should make sure we're taking into account concurrency when we run multiple creates with the same name.

Reply:
There's a card for this somewhere.  Hence we can ignore it for now

Reply:
Yup, [ARO-9674](https://issues.redhat.com/browse/ARO-9674)